builder/dockerfile/parser/parser.go

```diff
@@ -74,6 +93,26 @@ func parseLine(line string) (string, *Node, error) {
 		return "", nil, nil
 	}
 
+	// Handle the parser directive 'ESCAPE'. Parser directives must preceed
+	// any builder instruction, and cannot be repeated.
+	if tokenEscapeCommand.MatchString(strings.ToLower(line)) {
+		if builderInstructionProcessed == true {
```

Comment:
Since this doesn't have to be the first line, just the first command, it seems this approach still allows Dockerfiles with mixed escaping, e.g.:

    ESC\
    APE `

This may not matter (who would do this?), but it does mean that it's hard to write a truly correct parser for syntax highlighting.